### PR TITLE
Produce Debian 8 native package

### PR DESCRIPTION
**Pull request by wli3**
Since Debian is a branch from Ubuntu threat them the same

Tested on Debian 8 VM

### CODE REVIEW

File: build/FileExtensions.props [hunk 1]
```diff
@@ -7,10 +7,12 @@
       <InstallerExtension Condition=" '$(OSName)' == 'win' ">.msi</InstallerExtension>
       <InstallerExtension Condition=" '$(OSName)' == 'osx' ">.pkg</InstallerExtension>
       <InstallerExtension Condition=" '$(OSName)' == 'ubuntu' ">.deb</InstallerExtension>
+      <InstallerExtension Condition=" '$(OSName)' == 'debian' ">.deb</InstallerExtension>
```
Comment (livarcocc) [inline build/FileExtensions.props]: Should you just do an OR update there?

File: build/InstallerInfo.props [hunk 2]
```diff
@@ -5,8 +5,8 @@
     <CombinedFrameworkSdkHostInstallerFile>$(InstallerOutputDirectory)/$(ArtifactNameWithVersionCombinedHostHostFxrFrameworkSdk)$(BundleExtension)</CombinedFrameworkSdkHostInstallerFile>
 
     <SdkDebianIntermediateDirectory>$(IntermediateDirectory)/debian/sdk</SdkDebianIntermediateDirectory>
-    <LayoutDirectory Condition=" '$(OSName)' == 'ubuntu' ">$(SdkDebianIntermediateDirectory)/debianLayoutDirectory</LayoutDirectory>
-    <LayoutDirectory Condition=" '$(OSName)' != 'ubuntu' ">$(IntermediateDirectory)/layouts</LayoutDirectory>
+    <LayoutDirectory Condition=" '$(OSName)' == 'ubuntu' or '$(OSName)' == 'debian' ">$(SdkDebianIntermediateDirectory)/debianLayoutDirectory</LayoutDirectory>
```
Comment (livarcocc) [inline build/InstallerInfo.props]: NIT: let's be consistent on the casing for the conditionals here. Keep it all UPPER CASE. Like OR and AND.

File: build/publish/PublishDebian.targets [hunk 3]
```diff
@@ -19,7 +19,7 @@
     </DebianUploadJsonContent>
   </PropertyGroup>
 
-  <Target Name="PublishDebFilesToDebianRepo" Condition=" '$(OSName)' == 'ubuntu' AND '$(SkipPublishToDebianRepo)' != 'true' ">
+  <Target Name="PublishDebFilesToDebianRepo" Condition=" '( $(OSName)' == 'ubuntu' OR '$(OSName)' == 'debian' ) AND '$(SkipPublishToDebianRepo)' != 'true' ">
```
Comment (ellismg) [inline build/publish/PublishDebian.targets]: Just checking, does something need to change somewhere such that REPO_ID is set to something debian specific for the debian build (maybe this happens in official build definitions or something)?

Comment (wli3) [inline build/publish/PublishDebian.targets]: Yes, this is set in build definition on VSTS

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build
@dotnet-bot Test RHEL7.2 x64 Release Build

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

**Comment by livarcocc** [comment]
@MattGertz for approval.

**Comment by livarcocc** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build

**Comment by wli3** [comment]
@livarcocc  Could you forget to click the approve button

**Comment by wli3** [comment]
Find more problem: the name field is also

**Comment by wli3** [comment]
Testing on CI, there are more problems property "name" is absent for months as well.

I am copying the same property from Package to Publish.

Should I move it branding in ordering to share it?

**Comment by livarcocc** [comment]
What do you mean by move it branding in ordering to share?

File: build/publish/PublishDebian.targets [hunk 4]
```diff
@@ -1,11 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <ItemGroup>
-    <SdkInstallerFileItemGroup Include="$(SdkInstallerFile)" />
-  </ItemGroup>
+  <PropertyGroup>
+    <SdkDebianPackageName>dotnet-dev-$(SdkVersion)</SdkDebianPackageName>
```
Comment (livarcocc) [inline build/publish/PublishDebian.targets]: What are we using this for? It seems like we are not using this at all. Let's just remove it.

Comment (wli3) [inline build/publish/PublishDebian.targets]: It is used, that is the name of package name
for example the user should use

sudo apt-get install dotnet-dev-$(SdkVersion)

It was set in Packaging step. However, that prop cannot be shared with Publish.

**Comment by wli3** [comment]
> What do you mean by move it branding in ordering to share?

it is more like a code quality problem. This property is copy pasted from [there](https://github.com/dotnet/cli/blob/6ecc1fc6c3f58bde065413cc869fb7f97d296143/build/package/Installer.DEB.targets#L30)

To make them shared in code. I think the only place make a little sense is to move this property to [branding](https://github.com/dotnet/cli/blob/6ecc1fc6c3f58bde065413cc869fb7f97d296143/build/Branding.props#L22) 

Or we can just keep it this way.

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

**Comment by wli3** [comment]
Manuel test finished. Debian packaged showed up in apt-get feed
```
Package: dotnet-dev-2.0.0-preview2-006096
Priority: standard
Section: devel
Installed-Size: 185759
Maintainer: Microsoft <dotnetcore@microsoft.com>
Architecture: amd64
Version: 2.0.0-preview2-006096-1
Depends: libc6 (>= 2.14), dotnet-sharedframework-microsoft.netcore.app-2.0.0-preview2-25309-07
Filename: pool/main/d/dotnet-dev-2.0.0-preview2-006096/dotnet-sdk-debian-x64.2.0.0-preview2-006096.deb
Size: 75136206
MD5sum: b76686572f5aecbd037a8e40668a6960
SHA1: 03442ef36ef260dff2e571e0515bb7bdf943480d
SHA256: 38449082925e9947a95b7a68a900d041df00f3302b0fa339bf9a1f022b66031a
SHA512: 24a55ab0599d02e208a9c3a61467990a6381168098b88fb207718c0f7fcb6e9577353e63c04e190c76f59ec6f1513f435dd778c07b2218e3f68f49357119d520
Description: Microsoft .NET Core 2.0.0 - Preview 2 SDK
 .NET Core is a development platform that you can use to build command-line applications, microservices and modern websites. It is open source, cross-platform and is supported by Microsoft. We hope you enjoy using it! If you do, please consider joining the active community of developers that are contributing to the project on GitHub (https://github.com/dotnet/core). We happily accept issues and PRs.
Homepage: https://dotnet.github.io/core 
```
https://apt-mo.trafficmanager.net/repos/dotnet/dists/jessie/main/binary-amd64/Packages

**Comment by wli3** [comment]
**Customer scenario**

Debian/Ubuntu user install daily build version of dotnet CLI via command line.
sudo apt-get install <DebianPackageName>=<Version>

**Bugs this fixes**

https://github.com/dotnet/cli/issues/6113

**Workarounds, if any**

Use install script inside CLI repo

**Risk**

Debian, Ubuntu advance user cannot have native experience to install CLI

**Performance impact**

no

**Root cause analysis**

N/A

**How was the bug found?**

N/A

**Pull request merged**